File: Algorithm/Sorting Algorithms/bubbleSort.js
// WorestCase Time Complexity: O(n^2)
// BestCase Time Complexity: O(n)
// Average Time Complexity: O(n^2)
// Space Complexity: O(1)

function bubbleSort(arr) {
    let isSorted = false
    while (!isSorted) {
        isSorted = true
        for (let i = 0; i < arr.length - 1; i++) {
            if (arr[i] > arr[i + 1]) {
                isSorted = false
                let tempVar = arr[i]
                arr[i] = arr[i + 1]
                arr[i + 1] = tempVar
            }
        }
    }
    return arr
}

let output = bubbleSort([2, 3, 3, 1, 3, 4, 2, 4, 3, 3])
console.log(output)

File: Algorithm/Sorting Algorithms/bubble_sort.py
# Worst-case time complexity: O(n^2)
# Best-case time complexity: O(n)
# Average time complexity: O(n^2)
# Space complexity: O(1)

def bubble_sort(numbers):
    """
    Sort a list of numbers in ascending order using the bubble sort algorithm.

    Parameters:
    numbers (list): A list of numbers to be sorted.

    Returns:
    list: The sorted list of numbers.
    """
    is_sorted = False
    while not is_sorted:
        is_sorted = True
        for i, num in enumerate(numbers[:-1]):
            if num > numbers[i + 1]:
                numbers[i], numbers[i + 1] = numbers[i + 1], numbers[i]
                is_sorted = False
    return numbers

output = bubble_sort([3, 5, 2, 4, 33, 2, 1, 4, 10, 9, 5, 7, 6, 4, 3, 4])
print(output)